lib/syskit/cli/log_runtime_archive_main.rb

```diff
@@ -61,6 +61,42 @@ def archive(root_dir, target_dir)
                 archiver.process_root_folder
             end
 
+            desc "watch_transfer", "watches a dataset root folder \
+                                    and periodically performs transfer"
+            option :period,
+                   type: :numeric, default: 600, desc: "polling period in seconds"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            default_task def watch_transfer(src_dir, tgt_dir, server_params)
```

Comment:
> Also, here in watch_transfer I don't like to just call transfer because it creates an archiver and does socket allocation every time

What do you suggest? I decided to keep the names suggested in the shortcut story.

Reply:
She does need a target directory to transfer the files to.

Reply:
Also, I dont like the CLI interface for this command and the transfer. Server params shouldnt be a positional argument, it is a hash, it needs to be split into separate options  (host, port and anything else that would be included)

Reply:
> The tgt_dir parameter makes no sense for transfer and watch_transfer commands because they don't do the archiving

Right. I do plan on changing the initialize function so I can have the option to not pass a `target_dir` if not needed.

Reply:
> Also, I dont like the CLI interface for this command and the transfer. Server params shouldnt be a positional argument, it is a hash, it needs to be split into separate options (host, port and anything else that would be included)

Could you elaborate? I'm not sure if that's what you meant: having a variable for each value (host, port, etc.) and passing them as arguments to the `process_transfer` function? How should I deal with rubocop then (he doesn't like functions with more than 5 arguments), if that's indeed what you meant?

Reply:
I mean the interface one will use to interact with these commands. You cant pass a hash via command line, my suggestion is that the server_params become individual options.

Reply:
And you really should not write *anything* without understanding. This smells like blind copy/pasting.

What does `default_task` does ?

Reply:
> I mean on the interface one will use to interact with these commands (aka each one of the "default_task"s you created). You cant pass a hash via command line, my suggestion is that the server_params become individual options.

Ok. I'll still have the rubocop problem, but I guess there's no way to overcome this. Will disable the cop, then.